### ISSUE TITLE
Webdav connection fails under dvc 2.0.x

### ISSUE THREAD

**Issue by AnnKatrinBecker** [iterative/dvc#5588]
As long as I am using dvc 1.x I can push to the remote webdav storage.

But after an upgrade to dvc 2.0.4 I receive the following error:
`ERROR: failed to push data to the cloud - Unable to connect to WebDAV http:...`

webdav itself logs
`user <username>: authentication failure for "/...": Password Mismatch`

The user is send correctly, but the password does not seem to be picked up from config.local

**Comment by skshetry** [comment]
@AnnKatrinBecker, could you please post the command that's failing with the verbose output (please add `--verbose` at the end). 

Also, please post the complete `dvc doctor` output. Thanks.

**Comment by skshetry** [comment]
@AnnKatrinBecker, how are you adding the URL in the config? Does the URL need to be in https?
For example, my config looks something like following:
```
['remote "myremote"']
    url = webdavs://server.com/remote.php/dav/files/admin/dvc-cache/
```

Have you added it as `webdavs` or `webdav` instead of `http` or `https`?

**Comment by AnnKatrinBecker** [comment]
url is added in .dvc/config using `webdav://`

```
['remote "rawdata_remote"']
    url = webdav://storage...:8080/ml/rawdata
    user = username
```

**Comment by skshetry** [comment]
@AnnKatrinBecker, does it need to be `webdavs`? Is it an https connection? Maybe that's the problem?

Also, could you please try sending a curl request and see if it works there?
```console
$ curl -X PROPFIND http://<user>:<password>@storage...:8080/ml/rawdata -v | xmllint -format -
```

**Comment by AnnKatrinBecker** [comment]
should work via `webdav`
curl access is successful
if I downgrade back to dvc 1.11.16 dvc push works again

the password contains an @ might this lead to a resolving issue

**Comment by skshetry** [comment]
@AnnKatrinBecker, it might not be because of the password. 

I am not sure what's going on. It'd be great if you could debug it (with `dvc push --pdb`). Thanks. 🙂

**Comment by AnnKatrinBecker** [comment]
If I uncomment the following lines in dvc/fs/webdav.py I can push again

```
if not client.check("/"):
            raise WebDAVConnectionError(self.hostname)
```

**Comment by skshetry** [comment]
@AnnKatrinBecker, great find. So, your WebDAV server is not at the root of the URL, which is raising the issue. Our expectation here is wrong. Could you please try to change this to the following:
```python
if not client.check(self.path_info.path):
    raise WebDAVConnectionError(self.path_info.url)
```

If this works for you, we'd appreciate a pull request. Thanks again. 🙂

**Comment by skshetry** [comment]
Indeed, looks like this was introduced in 2.0 on b87e76aa844926a15244066fac4032c66fba22d1.

**Comment by skshetry** [comment]
@AnnKatrinBecker, this is now fixed in v2.0.16. Please update and let us know how it goes. 🙂

**Comment by AnnKatrinBecker** [comment]
> @AnnKatrinBecker, this is now fixed in v2.0.16. Please update and let us know how it goes. 🙂

After an update to the specified version, dvc push is working smoothly. Thank you very much!
